lib/str.js

```diff
@@ -935,6 +935,25 @@ const Str = {
         return CONST.SMS.E164_REGEX.test(phone);
     },
 
+    /**
+     * We validate mentions by checking if it's first character is an allowed character
+     * and by checking that we make sure it isn't inside other tags where mentions aren't allowed.
+     * For example, *@username@expensify.com* is a valid mention because we allow bold styling for it,
+     * but `@username@expensify.com` is not because we do not allow mentions within code
+     *
+     * @param {String} mention
+     * @returns {bool}
+     */
+    isValidMention(mention) {
```

Comment:
Dropping a note that this has caused https://github.com/Expensify/App/issues/20085
Missed an edge case where `test+1@gmail.com@gmail.com` was rendered as mention
![image](https://github.com/Expensify/expensify-common/assets/9059945/e0c492ec-bf22-4759-bbe3-86f2f90dd4cb)